src/pyhf/tensor/tensorflow_backend.py

```diff
@@ -11,12 +11,21 @@ class tensorflow_backend(object):
 
     def __init__(self, **kwargs):
         self.name = 'tensorflow'
+        self._mode = 'GPU'
         self.dtypemap = {
             'float': getattr(tf, kwargs.get('float', 'float32')),
             'int': getattr(tf, kwargs.get('int', 'int32')),
             'bool': tf.bool,
         }
 
+    @property
+    def mode(self):
+        return self._mode
+
+    @mode.setter
```

Comment:
To make this easier to deal with -- we shouldn't allow the mode to be changed after initialization. If the user wants to change the mode, they should supply a new backend...